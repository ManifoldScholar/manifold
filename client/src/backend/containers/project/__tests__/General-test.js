jest.mock("react-text-mask", () => () => "ReactTextMask");
import ProjectGeneralContainer from "../General";
import { project, route } from "./__fixtures__";

describe("backend/containers/project/General", () => {
  beforeEach(() => {
    testHelpers.startSession($dispatch, $user);
  });

  def("abilities", () => ({ update: true }));
  def("user", () => factory("user"));
  def("project", () => project($abilities));
  def("root", () => <ProjectGeneralContainer project={$project} />);

  it("matches the snapshot when rendered", () => {
    expect(mount($withApp($root)).html()).toMatchSnapshot();
  });

  it("does not render a null value", () => {
    expect(mount($withApp($root)).html()).not.toBeNull();
  });
});
