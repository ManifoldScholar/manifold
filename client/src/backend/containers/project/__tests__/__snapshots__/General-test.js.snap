// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backend Project General Container renders correctly 1`] = `
<section>
  <div>
    <form
      className="form-secondary"
      data-id="submit"
      onSubmit={[Function]}
    >
      <div
        className="form-section"
      >
        <header
          className="form-section-label"
        >
          <span>
            Title
          </span>
        </header>
        <div
          className="form-input-group"
        >
          <div
            className="form-input wide"
            style={Object {}}
          >
            <label
              className=""
              htmlFor="attributes[title]-text-input-1"
            >
              Title
            </label>
            <input
              aria-describedby="attributes[title]-text-input-1"
              id="attributes[title]-text-input-1"
              onChange={[Function]}
              placeholder="Enter Project Title"
              type="text"
              value="Rowan Test"
            />
          </div>
          <div
            className="form-input wide"
            style={Object {}}
          >
            <label
              className=""
              htmlFor="attributes[subtitle]-text-input-1"
            >
              Subtitle
            </label>
            <input
              aria-describedby="attributes[subtitle]-text-input-1"
              id="attributes[subtitle]-text-input-1"
              onChange={[Function]}
              placeholder="Enter Project Subtitle"
              type="text"
              value="World's Greatest Dog"
            />
          </div>
          <div
            className="form-input wide"
            style={Object {}}
          >
            <label
              className=""
              htmlFor="attributes[pendingSlug]-text-input-1"
            >
              Slug
            </label>
            <input
              aria-describedby="attributes[pendingSlug]-text-input-1"
              id="attributes[pendingSlug]-text-input-1"
              onChange={[Function]}
              placeholder="Enter Project Slug"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="form-section"
      >
        <header
          className="form-section-label"
        >
          <span>
            Visibility
          </span>
        </header>
        <div
          className="form-input-group"
        >
          <div
            className="form-input form-toggle-secondary"
          >
            <label
              className="form-input-heading toggle above"
              htmlFor="switch-input-1"
            >
              Draft Mode
            </label>
            <div
              className="toggle-indicator"
            >
              <div
                aria-pressed={false}
                className="boolean-primary"
                id="switch-input-1"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  className="screen-reader-text"
                >
                  Draft Mode
                </span>
              </div>
            </div>
            <span
              className="instructions"
            >
              A draft project is only visible to users who are able to modify it.
            </span>
          </div>
          <div
            className="form-input form-toggle-secondary"
          >
            <label
              className="form-input-heading toggle above"
              htmlFor="switch-input-1"
            >
              Featured
            </label>
            <div
              className="toggle-indicator"
            >
              <div
                aria-pressed={false}
                className="boolean-primary"
                id="switch-input-1"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  className="screen-reader-text"
                >
                  Featured
                </span>
              </div>
            </div>
            <span
              className="instructions"
            >
              Featured projects are highlighted on the home page.
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-section"
      >
        <header
          className="form-section-label"
        >
          <span>
            Other
          </span>
        </header>
        <div
          className="form-input-group"
        >
          <div
            className="form-input"
            style={Object {}}
          >
            <h4
              className="form-input-heading"
            >
              Publication Date
            </h4>
            <div
              className="form-date"
            >
              <div
                className="form-select input-month"
              >
                <i
                  aria-hidden="true"
                  className="manicon manicon-caret-down"
                />
                <select
                  onChange={[Function]}
                  value="NaN"
                >
                  <option />
                  <option
                    value={0}
                  >
                    January
                  </option>
                  <option
                    value={1}
                  >
                    February
                  </option>
                  <option
                    value={2}
                  >
                    March
                  </option>
                  <option
                    value={3}
                  >
                    April
                  </option>
                  <option
                    value={4}
                  >
                    May
                  </option>
                  <option
                    value={5}
                  >
                    June
                  </option>
                  <option
                    value={6}
                  >
                    July
                  </option>
                  <option
                    value={7}
                  >
                    August
                  </option>
                  <option
                    value={8}
                  >
                    September
                  </option>
                  <option
                    value={9}
                  >
                    October
                  </option>
                  <option
                    value={10}
                  >
                    November
                  </option>
                  <option
                    value={11}
                  >
                    December
                  </option>
                </select>
              </div>
              <div
                className="form-select input-day"
              >
                <i
                  aria-hidden="true"
                  className="manicon manicon-caret-down"
                />
                <select
                  onChange={[Function]}
                  value="NaN"
                >
                  <option />
                  <option>
                    1
                  </option>
                  <option>
                    2
                  </option>
                  <option>
                    3
                  </option>
                  <option>
                    4
                  </option>
                  <option>
                    5
                  </option>
                  <option>
                    6
                  </option>
                  <option>
                    7
                  </option>
                  <option>
                    8
                  </option>
                  <option>
                    9
                  </option>
                  <option>
                    10
                  </option>
                  <option>
                    11
                  </option>
                  <option>
                    12
                  </option>
                  <option>
                    13
                  </option>
                  <option>
                    14
                  </option>
                  <option>
                    15
                  </option>
                  <option>
                    16
                  </option>
                  <option>
                    17
                  </option>
                  <option>
                    18
                  </option>
                  <option>
                    19
                  </option>
                  <option>
                    20
                  </option>
                  <option>
                    21
                  </option>
                  <option>
                    22
                  </option>
                  <option>
                    23
                  </option>
                  <option>
                    24
                  </option>
                  <option>
                    25
                  </option>
                  <option>
                    26
                  </option>
                  <option>
                    27
                  </option>
                  <option>
                    28
                  </option>
                  <option>
                    29
                  </option>
                  <option>
                    30
                  </option>
                  <option>
                    31
                  </option>
                </select>
              </div>
              <div
                className="form-input"
              >
                ReactTextMask
              </div>
            </div>
          </div>
          <div
            className="form-input wide"
            style={Object {}}
          >
            <h4
              className="form-input-heading"
            >
              Subjects
            </h4>
            <nav
              className="has-many-list"
            >
              <div
                className="input-predictive"
              >
                <div
                  className="input"
                >
                  <input
                    aria-describedby={null}
                    aria-label="Add a Subject"
                    className="text-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Add a Subject"
                    type="text"
                    value=""
                  />
                  <button
                    aria-hidden="true"
                    className="manicon manicon-plus"
                    onClick={[Function]}
                    type="button"
                  />
                </div>
                <nav
                  className="predictive-list"
                >
                  <ul />
                </nav>
              </div>
              <ul
                className="bucket"
              >
                <li>
                  <div
                    className="association"
                  >
                    <h4
                      className="association-name"
                    >
                      Hip Hop
                    </h4>
                  </div>
                  <div
                    className="utility"
                  >
                    <button
                      className="manicon manicon-x"
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        className="screen-reader-text"
                      >
                        Remove 
                        Hip Hop
                         from the 
                        Subjects
                         list.
                      </span>
                    </button>
                  </div>
                </li>
              </ul>
            </nav>
          </div>
          <div
            className="form-input wide"
            style={Object {}}
          >
            <label
              className=""
              htmlFor="attributes[tagList]-tag-list-1"
            >
              Tags
            </label>
            <div
              className="input-predictive"
            >
              <div
                className="input"
              >
                <input
                  aria-describedby={null}
                  aria-label="Enter a Tag"
                  className="text-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Enter a Tag"
                  type="text"
                  value=""
                />
                <button
                  aria-hidden="true"
                  className="manicon manicon-plus"
                  onClick={[Function]}
                  type="button"
                />
              </div>
              <nav
                className="predictive-list"
              >
                <ul />
              </nav>
            </div>
            <div
              className="has-many-list"
            >
              <ul
                className="bucket empty"
              >
                <div
                  className="placeholder"
                >
                  None added
                </div>
              </ul>
            </div>
          </div>
          <div
            className="form-input avatar-builder wide"
            style={Object {}}
          >
            <div
              className="form-input avatar-builder wide"
            >
              <h4
                className="form-input-heading"
              >
                Project Thumbnail
              </h4>
              <div
                className="grid"
              >
                <div
                  className="section current"
                >
                  <span
                    aria-hidden="true"
                    className="label"
                  >
                    Current
                  </span>
                  <span
                    className="label screen-reader-text"
                  >
                    Current Project Thumbnail
                  </span>
                  <div
                    className="preview"
                  >
                    <svg
                      aria-label="Default Project Thumbnail"
                      className="project-thumb-placeholder project-thumb-placeholder--desktop project-thumb-placeholder--primary"
                      height={134}
                      viewBox="0 0 134 134"
                      width={134}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        fillRule="evenodd"
                      >
                        <g
                          className="project-thumb-placeholder__frame"
                          strokeWidth="1.5"
                          transform="translate(2 2)"
                        >
                          <polyline
                            points="124 6 124 124 6 124"
                          />
                          <polyline
                            points="130 12 130 130 12 130"
                          />
                          <polygon
                            points="0 118 118 118 118 0 0 0"
                          />
                        </g>
                        <polygon
                          className="project-thumb-placeholder__tile"
                          points="0 102 102 102 102 0 0 0"
                          transform="translate(10 10)"
                        />
                        <g
                          className="project-thumb-placeholder__illustration"
                          transform="translate(34 32)"
                        >
                          <g
                            transform="translate(47.557 2.968)"
                          >
                            <polyline
                              className="project-thumb-placeholder__illustration"
                              points=".271 7.606 .271 50.893 3.747 57.089 7.221 50.893 7.221 7.606"
                              strokeWidth="1.5"
                            />
                            <path
                              d="M7.22166456 50.8926203L.271601266 50.8926203M.27135443 3.82883544L.27135443 2.02693671C.27135443.936746835 1.1558481.0522531646 2.24603797.0522531646L5.24673418.0522531646C6.33774684.0522531646 7.22141772.936746835 7.22141772 2.02693671L7.22141772 3.82883544"
                              strokeWidth="1.5"
                            />
                            <polygon
                              points=".272 7.606 7.222 7.606 7.222 3.829 .272 3.829"
                              strokeWidth="1.5"
                            />
                            <path
                              d="M3.74655063,11.5277975 L3.74655063,46.9708987"
                              strokeWidth="1.5"
                            />
                            <polygon
                              points="2.183 54.3 3.747 57.089 5.311 54.3"
                            />
                          </g>
                          <g
                            strokeWidth="1.5"
                            transform="translate(0 26)"
                          >
                            <polygon
                              points="34.054 .598 .703 .598 .703 26.635 17.859 26.635 17.859 34.057 24.336 26.635 34.054 26.635"
                            />
                            <path
                              d="M28.3519304 7.9074557L6.30870253 7.9074557M28.3519304 18.7725759L6.30870253 18.7725759M28.3519304 13.3399747L6.30870253 13.3399747"
                            />
                          </g>
                          <g
                            strokeWidth="1.5"
                          >
                            <path
                              d="M27.3882848 11.8128038C27.3882848 14.7707152 24.9898671 17.1691329 22.0319557 17.1691329 19.0740443 17.1691329 16.6756266 14.7707152 16.6756266 11.8128038 16.6756266 8.85489241 19.0740443 6.45647468 22.0319557 6.45647468 24.9898671 6.45647468 27.3882848 8.85489241 27.3882848 11.8128038zM27.3882848 12.0765886L34.9537911 4.51108228C36.3196139 3.14608228 38.6546772 4.11285443 38.6546772 6.04475316M8.54099982 6.88421582L14.7943291.639221519C16.1601519-.726601266 18.4952152.240993671 18.4952152 2.17206962M11.4153165 12.3338734L12.5877848 11.8393797C13.52 11.4469114 14.5706962 11.4469114 15.5029114 11.8393797L16.6753797 12.3338734"
                            />
                            <path
                              d="M11.4153165,11.8128038 C11.4153165,14.7707152 9.01689873,17.1691329 6.05898734,17.1691329 C3.10107595,17.1691329 0.702658228,14.7707152 0.702658228,11.8128038 C0.702658228,8.85489241 3.10107595,6.45647468 6.05898734,6.45647468 C9.01689873,6.45647468 11.4153165,8.85489241 11.4153165,11.8128038 Z"
                            />
                          </g>
                        </g>
                      </g>
                    </svg>
                    <svg
                      aria-label="Default Project Thumbnail"
                      className="project-thumb-placeholder project-thumb-placeholder--mobile project-thumb-placeholder--primary"
                      height="100%"
                      viewBox="0 0 48 48"
                      width="100%"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        fillRule="evenodd"
                      >
                        <polygon
                          className="project-thumb-placeholder__tile"
                          fill="#CBF7E6"
                          points="0 42 42 42 42 0 0 0"
                        />
                        <g
                          className="project-thumb-placeholder__illustration"
                          transform="translate(9 8)"
                        >
                          <g
                            transform="translate(20.683 2.244)"
                          >
                            <polygon
                              points=".161 2.954 .161 21.013 1.612 23.598 3.06 21.013 3.06 2.954"
                            />
                            <path
                              d="M3.06081501,21.012797 L0.161633021,21.012797"
                            />
                            <path
                              d="M0.162011257,2.95560225 L3.06119325,2.95560225 L3.06119325,1.53299812 C3.06119325,0.980713374 2.613478,0.532998124 2.06119325,0.532998124 L1.16201126,0.532998124 C0.609726507,0.532998124 0.162011257,0.980713374 0.162011257,1.53299812 L0.162011257,2.95560225 Z"
                              strokeLinejoin="round"
                            />
                          </g>
                          <g
                            transform="translate(.195 11.463)"
                          >
                            <polygon
                              points="14.898 .419 .985 .419 .985 11.282 8.141 11.282 8.141 14.378 10.844 11.282 14.898 11.282"
                            />
                            <path
                              d="M12.5237854 3.46905816L3.32696735 3.46905816M12.5237854 8.00165403L3.32696735 8.00165403M12.5237854 5.73507242L3.32696735 5.73507242"
                            />
                          </g>
                          <g
                            transform="translate(.195 .195)"
                          >
                            <path
                              d="M11.8853223 5.12585966C11.8853223 6.3608015 10.8848871 7.36123677 9.64994522 7.36123677 8.41689456 7.36123677 7.41645929 6.3608015 7.41645929 5.12585966 7.41645929 3.89091782 8.41689456 2.89048255 9.64994522 2.89048255 10.8848871 2.89048255 11.8853223 3.89091782 11.8853223 5.12585966zM11.8853223 5.23573734L15.0417051 2.0793546C15.6109508 1.51010882 16.5849096 1.91293058 16.5849096 2.71857411M3.99814784 3.09718874L6.63067317.464663415C7.20181013-.106473546 8.17576886.2982394 8.17576886 1.10388293"
                            />
                            <path
                              d="M5.22155347 5.12585966C5.22155347 6.3608015 4.2211182 7.36123677 2.98617636 7.36123677 1.7531257 7.36123677.752690432 6.3608015.752690432 5.12585966.752690432 3.89091782 1.7531257 2.89048255 2.98617636 2.89048255 4.2211182 2.89048255 5.22155347 3.89091782 5.22155347 5.12585966zM5.22155347 5.34334559L5.71136961 5.13720675C6.09906191 4.97267392 6.53781614 4.97267392 6.92739962 5.13720675L7.41532458 5.34334559"
                            />
                          </g>
                        </g>
                        <g
                          className="project-thumb-placeholder__frame"
                          stroke="#828282"
                        >
                          <polyline
                            points="48 6 48 48 6 48"
                          />
                          <polyline
                            points="45 3 45 45 3 45"
                          />
                          <polygon
                            points="0 42 42 42 42 0 0 0"
                          />
                        </g>
                      </g>
                    </svg>
                    <span
                      className="screen-reader-text"
                    >
                      Default Project Thumbnail
                    </span>
                  </div>
                </div>
                <div
                  className="section color active"
                >
                  <span
                    aria-hidden="true"
                    className="label"
                  >
                    Default
                  </span>
                  <span
                    className="screen-reader-text"
                  >
                    Select a background color for the default thumbnail
                  </span>
                  <div
                    className="color-picker"
                  >
                    <div
                      className="colors"
                    >
                      <div
                        className="wrapper"
                      >
                        <label
                          className="form-toggle radio primary checked"
                          htmlFor="primary"
                        >
                          <span
                            className="screen-reader-text"
                          >
                            Primary Background Color
                          </span>
                          <input
                            checked={true}
                            id="primary"
                            onChange={[Function]}
                            type="radio"
                            value="primary"
                          />
                          <span
                            aria-hidden="true"
                            className="toggle-indicator"
                          >
                            <i
                              className="manicon manicon-check"
                            />
                          </span>
                        </label>
                        <label
                          className="form-toggle radio secondary"
                          htmlFor="secondary"
                        >
                          <span
                            className="screen-reader-text"
                          >
                            Secondary Background Color
                          </span>
                          <input
                            checked={false}
                            id="secondary"
                            onChange={[Function]}
                            type="radio"
                            value="secondary"
                          />
                          <span
                            aria-hidden="true"
                            className="toggle-indicator"
                          />
                        </label>
                        <label
                          className="form-toggle radio tertiary"
                          htmlFor="tertiary"
                        >
                          <span
                            className="screen-reader-text"
                          >
                            Tertiary Background Color
                          </span>
                          <input
                            checked={false}
                            id="tertiary"
                            onChange={[Function]}
                            type="radio"
                            value="tertiary"
                          />
                          <span
                            aria-hidden="true"
                            className="toggle-indicator"
                          />
                        </label>
                        <label
                          className="form-toggle radio quaternary"
                          htmlFor="quaternary"
                        >
                          <span
                            className="screen-reader-text"
                          >
                            Quaternary Background Color
                          </span>
                          <input
                            checked={false}
                            id="quaternary"
                            onChange={[Function]}
                            type="radio"
                            value="quaternary"
                          />
                          <span
                            aria-hidden="true"
                            className="toggle-indicator"
                          />
                        </label>
                        <label
                          className="form-toggle radio quinary"
                          htmlFor="quinary"
                        >
                          <span
                            className="screen-reader-text"
                          >
                            Quinary Background Color
                          </span>
                          <input
                            checked={false}
                            id="quinary"
                            onChange={[Function]}
                            type="radio"
                            value="quinary"
                          />
                          <span
                            aria-hidden="true"
                            className="toggle-indicator"
                          />
                        </label>
                        <label
                          className="form-toggle radio sentary"
                          htmlFor="sentary"
                        >
                          <span
                            className="screen-reader-text"
                          >
                            Sentary Background Color
                          </span>
                          <input
                            checked={false}
                            id="sentary"
                            onChange={[Function]}
                            type="radio"
                            value="sentary"
                          />
                          <span
                            aria-hidden="true"
                            className="toggle-indicator"
                          />
                        </label>
                      </div>
                    </div>
                    <div
                      aria-hidden="true"
                      className="default-description"
                    >
                      Select A Different Background Color
                    </div>
                  </div>
                </div>
                <div
                  className="section upload"
                >
                  <span
                    aria-hidden="true"
                    className="label"
                  >
                    Custom
                  </span>
                  <span
                    className="screen-reader-text"
                  >
                    Add a Custom Thumbnail Image
                  </span>
                  <div
                    className="form-input"
                  >
                    <div
                      className="form-input"
                      style={Object {}}
                    >
                      <react-dropzone
                        accepts={
                          Object {
                            "accepts": "image/*",
                            "extensions": "gif, jpeg, jpg, png",
                          }
                        }
                        className="form-dropzone style-square"
                        inputProps={
                          Object {
                            "aria-describedby": "upload-error-1",
                            "id": "upload-1",
                          }
                        }
                        multiple={false}
                        onDrop={[Function]}
                      >
                        <div
                          className="contents-empty"
                        >
                          <i
                            aria-hidden="true"
                            className="manicon manicon-cloud-up"
                          />
                          <div
                            className="message"
                          >
                            <p
                              className="primary"
                            >
                              <span
                                className="fake-link"
                                role="button"
                                tabIndex="0"
                              >
                                Upload a file
                              </span>
                               or 
                              <br />
                              drag and drop here
                              <br />
                            </p>
                            <p
                              className="secondary"
                            >
                              gif, jpeg, jpg, png
                            </p>
                          </div>
                        </div>
                      </react-dropzone>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-input submit wide"
      >
        <input
          className="button-secondary outlined"
          type="submit"
          value="Save Project"
        />
      </div>
    </form>
  </div>
</section>
`;
